### PR TITLE
Feature/stars

### PR DESCRIPTION
**Pull request by alexanderbock**
Adding the ability to remap SPECK parameters into internally used values to support SPECK files that have a different naming scheme from the DU files

### CODE REVIEW

**Comment by alexanderbock** [comment]
Closes https://github.com/OpenSpace/OpenSpace/issues/1592

File: modules/space/rendering/renderablestars.cpp [hunk 1]
```diff
@@ -120,6 +120,61 @@ namespace {
...
+    constexpr openspace::properties::Property::PropertyInfo MappingAbsMagnitudeInfo = {
+        "MappingAbsMagnitude",
+        "Mapping (absolute Magnitude)",
+        "The name of the variable in the speck file that is used as the absolute "
+        "magnitude variable"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo MappingAppMagnitudeInfo = {
+        "MappingAppMagnitude",
+        "Mapping (apparent Magnitude)",
```
Comment (WeirdRubberDuck) [inline modules/space/rendering/renderablestars.cpp:146]: Hm, wouldn't spell magnitude with a capital letter here (or above). looks a bit weird

File: modules/space/rendering/renderablestars.cpp [hunk 2]
```diff
@@ -120,6 +120,61 @@ namespace {
...
+        "The name of the variable in the speck file that is used as the apparent "
+        "magnitude variable"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo MappingVxInfo = {
+        "MappingVx",
+        "Mapping (vx)",
+        "The name of the variable in the speck file that is used as the star velocity in "
```
Comment (WeirdRubberDuck) [inline modules/space/rendering/renderablestars.cpp:154]: extra "in", for all the velocity directions. Should be either "... velocity along the x-axis" or "... velocity in the x direction"

File: modules/space/shaders/star_ge.glsl [hunk 3]
```diff
@@ -62,127 +56,164 @@ uniform dmat4 modelMatrix;
...
-const float SunTemperature = 5800.0f;
-const float SunAbsMagnitude = 4.83f;
+// FRAGILE
+// All of these values have to be synchronized with the values in the optionproperty
+const int SizeCompositionOptionAppBrightness = 0;
+const int SizeCompositionOptionLumSize = 1;
+const int SizeCompositionOptionLumSizeAppBrightness = 2;
+const int SizeCompositionOptionLumSizeAbsMagnitude = 3;
+const int SizeCompositionOptionLumSizeAppMagnitude = 4;
+const int SizeCompositionOptionLumSizeDistanceModulus = 5;
```
Comment (WeirdRubberDuck) [inline modules/space/shaders/star_ge.glsl:66]: Damn, those variable names ;) But much clearer now than before! 👍

File: modules/space/shaders/star_ge.glsl [hunk 4]
```diff
@@ -62,127 +56,164 @@ uniform dmat4 modelMatrix;
...
+
+double scaleForLuminositySizeAppBrightness(dvec3 dpos, float bv, float luminance, float absMagnitude) {
+    double luminosity = double(1.0 - luminance);
+    double distanceToStarInParsecs = trunc(length(dpos - eyePosition) / PARSEC);
+    double apparentBrightness = luminosity / distanceToStarInParsecs;
+    float L_over_Lsun = pow(2.51, SunAbsMagnitude - absMagnitude);
+    float temperature = bvToKelvin(bv);
+    float relativeTemperature = SunTemperature / temperature;
+    double starRadius = SunRadius * pow(relativeTemperature, 2.0) * sqrt(L_over_Lsun);
+    return (lumCent * (luminance + 5E9) + (radiusCent * starRadius) + (brightnessCent * apparentBrightness * 5E15)) * pow(10.0, magnitudeExponent);
```
Comment (WeirdRubberDuck) [inline modules/space/shaders/star_ge.glsl:102]: consider breaking this line?

File: modules/space/shaders/star_ge.glsl [hunk 5]
```diff
@@ -62,127 +56,164 @@ uniform dmat4 modelMatrix;
...
-    
-    gl_Position = bottomLeftVertex;
-    psfCoords   = vec2(-1.0, -1.0);
+    vec4 upperLeft = z_normalization(
+        vec4(cameraViewProjectionMatrix * dvec4(dpos.xyz + scaledUp - scaledRight, dpos.w))
+    );
+
+    gs_screenSpaceDepth  = lowerLeft.w;
+
+    // Build primitive
```
Comment (WeirdRubberDuck) [inline modules/space/shaders/star_ge.glsl:202]: extra whitespaces at the end here

**Review by WeirdRubberDuck — CHANGES_REQUESTED** [review]
Nice feature! Looks great, but as the `DataMapping` property is not optional it needs ot be added to all the other Assets with Renderables stars to avoid errors like this: 
image

So, the following files: apogee.asset, galah.asset, cluster.asset, denver.asset

File: modules/space/rendering/renderablestars.cpp (hunk 1 above)
Comment (alexanderbock) [reply modules/space/rendering/renderablestars.cpp:146]: Copy&Paste error

File: modules/space/shaders/star_ge.glsl (hunk 3 above)
Comment (alexanderbock) [reply modules/space/shaders/star_ge.glsl:66]: I'd be happy if GLSL would have some enum construct instead ;)

**Review by WeirdRubberDuck — APPROVED** [review]

**Pull request merged**